File: thbigmatome/app/models/cost.rb
class Cost < ApplicationRecord
  has_many :cost_players, dependent: :destroy
  has_many :players, through: :cost_players

  validates :name, presence: true
  validates :start_date, presence: true

  def self.current_cost
    Cost.where(end_date: nil).first
  end
end

File: thbigmatome-front/src/types/cost.ts
export interface Cost {
  id: number;
  name: string;
  start_date: string;
  end_date: string;
  normal_cost: number | null;
  relief_only_cost: number | null;
  pitcher_only_cost: number | null;
  fielder_only_cost: number | null;
  two_way_cost: number | null;
}